File: backend-dotnet/Application/Interfaces/IElasticService.cs
using Application.Common;

namespace Application.Interfaces
{
    public interface IElasticService
    {
        Task IndexUserAsync(UserIndex userIndex);
        Task IndexPostAsync(PostIndex postIndex);
        Task IndexTagAsync(TagIndex tagIndex);
        Task UpdateUserAsync(UserIndex userIndex);
        Task UpdatePostAsync(PostIndex postIndex);
        Task DeletePostAsync(string postId);
        Task<List<PostIndex>> SearchPostsAsync(string query);
        Task<List<UserIndex>> SearchUsersAsync(string query, int size = 10);
        Task<List<TagIndex>> SearchTagsAsync(string query, int size = 3);
    }
}

File: backend-java/src/main/java/com/jerrygram/application/interfaces/IElasticService.java
package com.jerrygram.application.interfaces;

import com.jerrygram.application.common.PostIndex;
import com.jerrygram.application.common.TagIndex;
import com.jerrygram.application.common.UserIndex;

import java.util.List;

public interface IElasticService {
    
    void indexUser(UserIndex userIndex);
    
    void indexPost(PostIndex postIndex);
    
    void indexTag(TagIndex tagIndex);
    
    void updateUser(UserIndex userIndex);
    
    void updatePost(PostIndex postIndex);
    
    void deletePost(String postId);
    
    List<PostIndex> searchPosts(String query);
    
    List<UserIndex> searchUsers(String query, int size);
    
    List<TagIndex> searchTags(String query, int size);
}